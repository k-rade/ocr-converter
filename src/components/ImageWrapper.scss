.image-wrapper {
  display: flex;
  width: 500px;
  height: 500px;
  border: 3px dashed gray;
  align-items: center;

  &-text {
    display: flex;
    flex-direction: column;
    text-align: center;
    margin: 0 2rem;
  }

  img {
    width: 500px;
    height: 500px;
    position: relative;
  }
}
