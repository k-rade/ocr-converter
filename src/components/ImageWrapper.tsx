import { ReactComponent as ImagePreviewIcon } from "../icons/image-preview.svg";

import "./ImageWrapper.scss";

interface Props {
  imageUrl?: string;
}

const ImageWrapper = ({ imageUrl = "" }: Props): JSX.Element => {
  return (
    <div className="image-wrapper">
      {imageUrl ? (
        <img src={imageUrl} alt="can not load image" />
      ) : (
        <div className="image-wrapper-text">
          <ImagePreviewIcon />
          <div>
            <h3>Image preview</h3>
            <span className="image-wrapper-text-desc">
              You should see your image here. <br />
              If image can not be displayed, <br /> there might be an issue with
              image URL.
            </span>
          </div>
        </div>
      )}
    </div>
  );
};

export default ImageWrapper;
