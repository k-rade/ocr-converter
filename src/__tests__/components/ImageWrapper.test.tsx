import { render } from "@testing-library/react";
import ImageWrapper from "../../components/ImageWrapper";

describe("ImageWrapper", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<ImageWrapper />);

    expect(baseElement).toMatchSnapshot();
  });
});
