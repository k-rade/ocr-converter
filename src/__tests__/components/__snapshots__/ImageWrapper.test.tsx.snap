// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWrapper should render 1`] = `
<body>
  <div>
    <div
      class="image-wrapper"
    >
      <div
        class="image-wrapper-text"
      >
        <svg>
          image-preview.svg
        </svg>
        <div>
          <h3>
            Image preview
          </h3>
          <span
            class="image-wrapper-text-desc"
          >
            You should see your image here. 
            <br />
            If image can not be displayed, 
            <br />
             there might be an issue with image URL.
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
